seapath_ros_driver/include/seapath_ros_driver.hpp

```diff
@@ -79,139 +50,133 @@ namespace seapath
       /**
        * @brief Sets up the socket for communication.
        *
-       * This function initializes the socket for UDP communication with the specified IP address and port number.
+       * This function initializes the socket for UDP communication with the specified IP address and port number from ROS params.
        *
-       * @param UDP_IP The IP address to bind the socket to.
-       * @param UDP_PORT The port number to bind the socket to.
        */
-      void SetupSocket(std::string UDP_IP, uint16_t UDP_PORT);
+      void setup_socket();
+
 
-      /**
-       * @brief Timer callback function that retrieves KMBinaryData, then publishes it.
-       */
-      void timer_callback();
 
       /**
-       * @brief Publish method that calls various publishers.
+       * @brief Sets the initial origin from the world frame and initiates the timer callback once origin is set.
+       *
        *
-       * @param data The KMBinaryData object to be published.
        */
-      void publish(KMBinaryData data);
+      void initial_setup();
+
 
       /**
-       * @brief Process the new data received from the socket
-       *
-       * @param data The new data
+       * @brief Timer callback function that retrieves KMBinaryData, then publishes it.
        */
-      void process_kmbinary_data(std::vector<uint8_t> data);
+      void timer_callback(); 
 
       /**
-       * @brief Parses the new data received from the socket
+       * @brief Publishes static transform from world NED to world SEU to use for foxglove visualization.
        *
-       * @param data Variable for storing the new data
-       * @return KMBinaryData - the new data. Is only updated with new data IF the socket is connected. That is done by checking the socket.connected boolean.
        */
-      KMBinaryData parse_kmbinary_data(std::vector<uint8_t> data);
+      void publish_static_tf(const rclcpp::Time& time) const;
```

Comment:
maybe rename this function, static tf could be lots of transforms

Reply:
renamed to publish_foxglove_vis_frame